File: src/main/java/com/thebluehats/server/core/modules/PluginModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;

import com.thebluehats.server.core.TheBlueHatsServerPlugin;
import org.bukkit.plugin.java.JavaPlugin;

public class PluginModule extends AbstractModule {
    private final TheBlueHatsServerPlugin plugin;

    public PluginModule(TheBlueHatsServerPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    protected void configure() {
        bind(JavaPlugin.class).toInstance(plugin);
        bind(TheBlueHatsServerPlugin.class).toInstance(plugin);
    }
}

File: src/main/kotlin/com/thebluehats/server/core/modules/PluginModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.thebluehats.server.core.TheBlueHatsServerPlugin
import org.bukkit.plugin.java.JavaPlugin

class PluginModule(private val plugin: TheBlueHatsServerPlugin) : AbstractModule() {
    override fun configure() {
        bind(JavaPlugin::class.java).toInstance(plugin)
        bind(TheBlueHatsServerPlugin::class.java).toInstance(plugin)
    }
}